### PR TITLE
add json file to config unittest

### PR DESCRIPTION
**Pull request by lightszero**
add json file to config unittest.

### CODE REVIEW

File: Neo.Compiler.MSIL.UnitTests/TestJsons/test_newarray.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "tests": {
+    "newarray": {
+      "build": [ "./TestClasses/Contract1.cs" ],
+
+      "testmethod": "execute",
+      "entryscript": "./TestClasses/Contract1.cs",
+      "testparams": [
+        "(str)test",
```
Comment (shargon) [inline Neo.Compiler.MSIL.UnitTests/TestJsons/test_newarray.json]: {"type":"String","value":"test"}

File: Neo.Compiler.MSIL.UnitTests/TestJsons/test_newarray.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        []
+      ],
+      "testresult": "(bytes)01020304"
```
Comment (shargon) [inline Neo.Compiler.MSIL.UnitTests/TestJsons/test_newarray.json]: same as neo-vm test will be better

File: Neo.Compiler.MSIL.UnitTests/TestJsons/test_newarray.json (hunk 1 above)
Comment (lightszero) [reply Neo.Compiler.MSIL.UnitTests/TestJsons/test_newarray.json]: I think this

[
   "(str)transfer",
   ["(address)fromaddrxxxx", 
     "(address)toaddrxxxx",
     12345555]
]

is more clearfiy that

[
     {"type":"string","value":"transfer"},
     [
          {"type":"address","value":"fromaddrxxxx"},
          {"type":"address","value":"toaddrxxxx"},
          {"type":"number","value":12345555},
    ]
]

more params,more clearfiy

Comment (shargon) [reply Neo.Compiler.MSIL.UnitTests/TestJsons/test_newarray.json]: Is more small, but for parsers is better the biggest one. But we should decide what we should use, but the same in all projects.

Comment (lightszero) [reply Neo.Compiler.MSIL.UnitTests/TestJsons/test_newarray.json]: small and keep same structure with the stackitem.but both way is fine for me.
@neo-project/core what do you think?

**Comment by erikzhang** [comment]
What's the difference between using JSON files and just using `Directory.GetFiles()` to get all *.cs files from `TestClasses` directory?

**Comment by lightszero** [comment]
> What's the difference between using JSON files and just using `Directory.GetFiles()` to get all *.cs files from `TestClasses` directory?

json config the testparam and testresult,not for build only.

File: Neo.Compiler.MSIL.UnitTests/Utils/TestEngine.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    class TestEngine : ExecutionEngine
+    {
+        IDictionary<string, BuildScript> contracts;
+        public TestEngine(IDictionary<string, BuildScript> contracts)
+        {
+            this.contracts = contracts;
+        }
+        public RandomAccessStack<StackItem> ExecuteTestCase(BuildScript entryscript, StackItem[] _params)
+        {
+            //var engine = new ExecutionEngine();
```
Comment (lock9) [inline Neo.Compiler.MSIL.UnitTests/Utils/TestEngine.cs]: Remove dead code

File: Neo.Compiler.MSIL.UnitTests/Utils/base58.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+using System.Linq;
+using System.Numerics;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ThinNeo.Cryptography.Cryptography
+{
+    public static class Base58
+    {
```
Comment (lock9) [inline Neo.Compiler.MSIL.UnitTests/Utils/base58.cs]: Is this class really required? Can't we use some other class that already exists? Or does this class does something different?

File: Neo.Compiler.MSIL.UnitTests/Utils/BuildScript.cs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            this.modIL = new ILModule(log);
+            try
+            {
+                modIL.LoadModule(fs, fspdb);
+            }
+            catch (Exception err)
+            {
+                log.Log("LoadModule Error:" + err.ToString());
+                this.Error = err;
+                return;
```
[the remark this answers is not in the document]
Comment (lightszero) [reply Neo.Compiler.MSIL.UnitTests/Utils/BuildScript.cs]: This pr is too old,I had read  your suggest.
I will fix this pr soon.

[the remark this answers is not in the document]
Comment (lock9) [reply Neo.Compiler.MSIL.UnitTests/Utils/BuildScript.cs]: Sorry for the inconvenience @lightszero, I can assure you things will start moving faster.

**Comment by shargon** [comment]
Conflicts

**Comment by lock9** [comment]
@lightszero  we can close this, right? #121

**Comment by meevee98** [comment]
I've tested this pull request and it's working as expected.
The tests are available [here](https://simpli.testquality.com/project/9154/plan/15601/run/122277/result/1678126).

**Pull request closed without merging**